src/scripts/modules/destinations.js

```diff
@@ -0,0 +1,134 @@
+  bucketlist.addEventListener("click", addBucketlist);
+  all.addEventListener("click", showAll);
+  document.addEventListener("DOMContentLoaded", setActive);
+  document.addEventListener("scroll", shrinkNav);
+  window.addEventListener("scroll", restoreNav);
+}
+
+function renderDestination(destination) {
+  // create destination wrapper
+  const elDestination = document.createElement("figure");
+  elDestination.classList.add("img-container");
+
+  // add the correct class to elDestination wrapper
+  if (destination.visited === true) {
+    elDestination.classList.add("visited");
+  } else {
+    elDestination.classList.add("bucketlist");
+  }
+
+  // create img element
+  const img = document.createElement("img");
+  img.style.objectFit = "cover"; // TODO - Prefer adding a CSS class instead.
+  img.src = destination.photo;
+  // add img to destination
+  elDestination.appendChild(img);
+
+  //set img captions
+  const figcaption = document.createElement("figcaption");
+  figcaption.classList.add("caption");
+  figcaption.insertAdjacentHTML(
+    "beforeend",
+    `<h2 class='caption__city'>${destination.city}</h2>`
+  );
+  figcaption.insertAdjacentHTML(
+    "beforeend",
+    `<h3 class='caption__country'>${destination.country}</h3>`
+  );
+  // add captions to destination
+  elDestination.appendChild(figcaption);
+
+  // Return the destination HTML
+  return elDestination;
+}
+
+function render() {
+  // Create DOM elements for each destination
+  destinationsList.forEach((destination) => {
+    const elDestination = renderDestination(destination);
+    gallery.appendChild(elDestination);
+  });
+}
+
+function setActive() {
+  buttons.forEach((button) => {
+    button.addEventListener("click", (event) => {
+      const current = event.target;
+
+      if (activeBtn) {
+        activeBtn.classList.remove("active-button");
+        activeBtn.setAttribute("aria-pressed", false);
+      }
+
+      current.setAttribute("aria-pressed", true);
+      current.classList.add("active-button");
+
+      activeBtn = current;
+    });
+  });
+  //set visited as the default active button on page load
+  document.querySelector(".button__visited").click();
+}
+
+function addVisited() {
+  gallery.classList.add("visited");
+  gallery.classList.remove("bucketlist");
+}
+
+function addBucketlist() {
+  gallery.classList.add("bucketlist");
+  gallery.classList.remove("visited");
+}
+
+function showAll() {
+  gallery.classList.remove("visited");
+  gallery.classList.remove("bucketlist");
+}
+
+//shrink header on scroll
+function shrinkNav() {
+  const navbar = document.querySelector(".navbar");
+  navbar.classList.add("shrink");
+}
+
+// restore nav size on scroll up
+function restoreNav() {
+  const scrollPos = 0;
+  if (document.body.getBoundingClientRect().top === scrollPos) {
+    document.querySelector(".navbar").classList.remove("shrink");
+  }
+}
+
+function init() {
+  cacheDom();
+  bindEvents();
+  render();
+}
+
+init();
```

Comment:
You don't need to call `init()` here. The initialization should only be done in the file that imports this module, as you did. Otherwise you'll init this module twice, and that will lead to unexpected bugs

Reply:
Got it